prime-dotnet-webapi/Services/OrganizationService.cs

```diff
@@ -60,6 +63,7 @@ public async Task<IEnumerable<OrganizationSearchViewModel>> GetOrganizationsAsyn
                         o => o.Sites.Select(s => s.PEC))
                     .Containing(searchOptions.TextSearch)
                 )
+                .Include(o => o.Claims)
```

Comment:
This might be surprising, but you don't need to `.Include()` the Claims here for `HasClaim` to appear on the View Model. I think it's worth knowing why it works this way so you can have a better understanding of how AutoMapper + Entity Framework works.

As a baseline, instead of using AutoMapper, we could project to a View Model by doing a `.Select()` statement. Entity Framework takes that `.Select()` statement and uses it to generate SQL; so the `.ProjectTo()` could be re-written without AutoMapper by doing:
```C#
.Select(org => new OrganizationListViewModel
{
    Id = org.Id,
    Name = org.Name,
    ...
    HasClaim = org.Claims.Any()
})
```
EF knows that the `Claims` property is on another table, so it will automatically generate the SQL it needs. Sometimes this is a join, but in this case the SQL is something like 
```SQL
SELECT EXISTS (
    SELECT 1
    FROM "OrganizationClaim" AS oc
    WHERE o."Id" = oc."OrganizationId") AS "HasClaim"
FROM "Organization" AS o
```
Adding in AutoMapper: the mapping configuration, by default, copies Properties if their name and type match. So AM sees that the View Model has a Property called `HasClaim`, and the Model has a matching property, so it generates
```C#
.Select(org => new OrganizationListViewModel
{
    Id = org.Id,
    Name = org.Name,
    ...
    HasClaim = org.HasClaim
})
```
But this wont work; EF doesn't know how to look inside the `HasClaim` property since it is effectively a black box. This is where Delegate Decompiler comes in: calling `.DecompileAsync()` turns all the `[Computed]` Properties into their expressions, making the above `.Select()` into:
```C#
.Select(org => new OrganizationListViewModel
{
    Id = org.Id,
    Name = org.Name,
    ...
    HasClaim = org.Claims.Any()
})
```
as we had before.

In contrast, using `.Include()` does an eager load and will, usually, do a full SELECT * + JOIN to the other table. It is kinda in direct opposition to the View Models we are building: VMs declare the (often flattened) subset of Properties the we are interested in, while `.Include()` loads everything. In general, if we are projecting to a VM, we don't need (or want) to use `.Include()`.

Reply:
FYI @james-hollinger , I added a link to this conversation into my OneNote collection.  Plus we are 1 more comment closer to triple digit comments on this Pull Request!  😊